File: android/src/new/java/com/margelo/nitro/rive/DeprecationWarning.kt
package com.margelo.nitro.rive

object DeprecationWarning {
    private val warned = mutableSetOf<String>()

    fun warn(method: String, replacement: String) {
        if (warned.add(method)) {
            RiveLog.w("Deprecation",
                "'$method' is deprecated and blocks the calling thread. Use '$replacement' instead.")
        }
    }
}

File: ios/DeprecationWarning.swift
enum DeprecationWarning {
    private static var warned = Set<String>()

    static func warn(_ method: String, replacement: String) {
        guard !warned.contains(method) else { return }
        warned.insert(method)
        RiveLog.w("Deprecation",
            "'\(method)' is deprecated and blocks the JS thread. Use '\(replacement)' instead.")
    }
}
